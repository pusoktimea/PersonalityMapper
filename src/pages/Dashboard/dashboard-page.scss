.dashboard-page {
    .persmap-icon {
        margin-right: 10px;
        margin-bottom: 15px
    }
    .persmap-panel_body {
      a {
        color: #6B6B6B;
        text-decoration: none;
      }
      a:hover {
        color: #6B6B6B;
        text-decoration: underline;
      }
    }
}

#chartjs-tooltip {
  line-height: normal;
  opacity: 1;
  padding: .3em;
  pointer-events: none;
  position: fixed;
  text-align: center;
  transform: translate(-50%, 0);
  transition: all .1s ease;
}
