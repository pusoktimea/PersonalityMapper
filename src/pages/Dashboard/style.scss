@import '~compass-mixins/lib/compass';
@import "../../styles/_colors.scss";

$baseClass: '.activate-dashboard';

#{$baseClass} {
    &_button {
        margin-bottom: 10px;
        &_style {
            width: 175px;
            font-size: 12px;
            .activate-icon {
                margin-right: 5px;
            }
        }
    }
    
}

.activate-row .activate-panel_body {
    text-align: center;
}