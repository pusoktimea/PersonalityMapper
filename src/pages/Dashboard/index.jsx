import React, {PureComponent, Fragment} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import cookie from 'js-cookie';
import {Link} from 'react-router-dom';

import {doGet} from '../../utils/APIUtils';

import {Doughnut, Bar} from 'react-chartjs-2';

import Row from 'components/Grid/Row';
import Column from 'components/Grid/Column';
import Panel from 'components/Panel';
import Icon from 'components/Icon';
import Loader from 'components/Loader';

import './dashboard-page.scss';

class Dashboard extends PureComponent {
  static propTypes = {
    isTablesLoaded: PropTypes.bool,
    onGetTables: PropTypes.func,
    tables: PropTypes.object,
    isSideBarMinimised: PropTypes.bool
  };

  constructor(props) {
    super(props);
    this.state = {
      allTeams: [],
      allUsers: [],
      allPerstype: {},
      persInTeam: [],
      persTypeInTeam: [],
      teamMemberNames: [],
      team: null
    };
  }

  componentWillMount() {
    const loggedInUser = cookie.get('loggedInUser');
    doGet(`userInfo/${loggedInUser}`).then((response) => {
      this.setState({team: response.data.team});
      doGet(`perstype/${response.data.team}`).then((response) => {
        this.setState({persInTeam: response.data});
        const teamMemberNames = response.data.map(item => (item.profile.name));
        this.setState({teamMemberNames: teamMemberNames});
        const persTypeInTeam = response.data.map(item => (item.profile.persType));
        const teamPersTypeCounter = persTypeInTeam.reduce((persTypeCount, currentPerstype) =>  {
          if(typeof persTypeCount[currentPerstype] !== 'undefined'){
            persTypeCount[currentPerstype]++;
            return persTypeCount;
          } else {
            persTypeCount[currentPerstype] = 1;
            return persTypeCount;
          }
        }, {});
        this.setState({persTypeInTeam: teamPersTypeCounter});
      });
    });
    doGet('allTeams').then((response) => {
      const teamCounter = response.data.reduce((teamCount, currentTeam) =>  {
        if(typeof teamCount[currentTeam.team] !== 'undefined'){
          teamCount[currentTeam.team]++;
          return teamCount;
        } else {
          teamCount[currentTeam.team] = 1;
          return teamCount;
        }
      }, {});
      this.setState({allTeams: teamCounter});
    });
    doGet('allUsers').then((response) => {
      this.setState({allUsers: response.data});
    });
    doGet('allPerstype').then((response) => {
      const allPersType = response.data.map(item => (item.profile.persType));
      const persTypeCounter = allPersType.reduce((persTypeCount, currentPerstype) =>  {
        if(typeof persTypeCount[currentPerstype] !== 'undefined'){
          persTypeCount[currentPerstype]++;
          return persTypeCount;
        } else {
          persTypeCount[currentPerstype] = 1;
          return persTypeCount;
        }
      }, {});
      this.setState({allPerstype: persTypeCounter});
    });
  }

  managerView = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  barChart() {
    const persTypes = Object.keys(this.state.allPerstype);
    const persTypeNumbers = Object.values(this.state.allPerstype);
    persTypeNumbers.push(0);
    return {
      labels: persTypes,
      datasets: [
        {
          label: 'MBTI Personality Type - Test Results',
          backgroundColor: 'rgba(255,99,132,0.2)',
          borderColor: 'rgba(255,99,132,1)',
          borderWidth: 1,
          hoverBackgroundColor: 'rgba(255,99,132,0.4)',
          hoverBorderColor: 'rgba(255,99,132,1)',
          data: persTypeNumbers
        }
      ]
    };
  }

  doughnutData(team) {
    const persTypes = Object.keys(this.state.persTypeInTeam);
    const persTypeSum = Object.values(this.state.persTypeInTeam);
    return {
      labels: persTypes,
      datasets: [{
        data: persTypeSum,
        backgroundColor: [
          '#FF6384',
          '#36A2EB',
          '#9bc898',
          '#ffe766'
        ],
        hoverBackgroundColor: [
          '#FF6384',
          '#36A2EB',
          '#9bc898',
          '#ffe766'
        ]
      }]
    };
  }

  render() {
    const {
      isSideBarMinimised
    } = this.props;
    const baseClass = 'main-content';
    return (
      <div className={cx('dashboard-page', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Dashboard</h2>
        <Row columnCount={3}>
          {
            this.state.team === null ?
              <Loader /> :
              <Fragment>
                <Column
                  style={{
                    textAlign: 'center'
                  }}
                  width={6}
                >
                  {
                    this.state.team === 'Management' ?
                      <div>
                        <h2>
                          Personality types in your company
                        </h2>
                        <Bar data={this.barChart()} />
                      </div> :
                      <div>
                        <h2>
                          Personality types in team {this.state.team}
                        </h2>
                        <Doughnut data={this.doughnutData(this.state.team)} />
                      </div>
                  }
                </Column>
                {
                  this.state.team === 'Management' ?
                    <Column width={3}>
                      <Panel title="Teams">
                        {
                          Object.keys(this.state.allTeams).map((item, index) => (
                            <div key={index}>
                              <Icon icon="users" />
                              <span>
                                {item}
                              </span>
                            </div>
                          ))
                        }
                      </Panel>
                    </Column> :
                    <Column width={3}>
                      <Panel title="My Team">
                        {
                          this.state.persInTeam.map((item, index) => (
                            <div key={index}>
                              <Icon icon="user" />
                              <Link to={`/profile/${item.username}`}>
                                {item.profile.name} - {item.profile.persType}
                              </Link>
                            </div>
                          ))
                        }
                      </Panel>
                    </Column>
                }
                <Column width={3}>
                  <Panel title="My Company">
                    {
                      this.state.allUsers.map((item, index) => (
                        <div key={index}>
                          <Icon icon="user" />
                          <Link to={`/profile/${item.username}`}>
                            {item.profile.name}
                          </Link>
                        </div>
                      ))
                    }
                  </Panel>
                </Column>
              </Fragment>
          }
        </Row>
      </div>
    );
  }
}

export default Dashboard;
