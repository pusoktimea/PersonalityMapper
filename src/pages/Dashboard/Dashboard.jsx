import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Loader from 'components/Loader';
import SearchForm from 'components/SearchForm';
import Panel from 'components/Panel';
import Row from 'components/Grid/Row';
import Column from 'components/Grid/Column';
import Button from 'components/Button';
import Icon from 'components/Icon';

import './style.scss';

class Dashboard extends PureComponent {
  static propTypes = {
    isSearchInProgress: PropTypes.bool.isRequired,
    isSearchComplete: PropTypes.bool.isRequired,
    onDoSearch: PropTypes.func.isRequired,
    searchResults: PropTypes.array.isRequired,
    doResetSearch: PropTypes.func.isRequired,
    location: PropTypes.object.isRequired,
    isTablesLoaded: PropTypes.bool.isRequired,
    onGetTables: PropTypes.func.isRequired,
    tables: PropTypes.object.isRequired,
    isSideBarMinimised: PropTypes.bool
  };

  componentDidMount() {
    if (!this.props.isTablesLoaded) {
      this.props.onGetTables();
    }
  }

  render() {
    const {
      isSearchInProgress,
      isSearchComplete,
      onDoSearch,
      searchResults,
      doResetSearch,
      location,
      isSideBarMinimised,
      isTablesLoaded,
      tables: {list}
    } = this.props;
    const baseClass = 'main-content';

    return (
      <div className={cx('activate-dashboard', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Dashboard</h2>
        {
          !isTablesLoaded ?
            <Loader /> :
            <div>
              <SearchForm
                isSearchInProgress={isSearchInProgress}
                isSearchComplete={isSearchComplete}
                onDoSearch={onDoSearch}
                searchResults={searchResults}
                doResetSearch={doResetSearch}
                location={location}
              />
              <Row columnCount={3}>
                <Column width={4}>
                  <Panel title="My reports">
                    <div className="activate-dashboard_button">
                      <Button
                        href="/reports/my_program"
                        className="activate-dashboard_button_style"
                        theme="default"
                      >
                        My Programs
                      </Button>
                    </div>
                    <div className="activate-dashboard_button">
                      <Button
                        href="/reports/my_goal"
                        className="activate-dashboard_button_style"
                        theme="default"
                      >
                        My Goals
                      </Button>
                    </div>
                    <div className="activate-dashboard_button">
                      <Button
                        href="/vendor_report"
                        className="activate-dashboard_button_style"
                        theme="default"
                      >
                        Vendor Reports
                      </Button>
                    </div>
                  </Panel>
                </Column>

                <Column width={4}>
                  <Panel title="My shortcuts">
                    <div className="activate-dashboard_button">
                      <Button
                        className="activate-dashboard_button_style"
                        theme="primary"
                      >
                        <Icon icon="plus" />
                        Add Programs
                      </Button>
                    </div>
                    <div className="activate-dashboard_button">
                      <Button
                        className="activate-dashboard_button_style"
                        theme="primary"
                      >
                        <Icon icon="plus" />
                        Add Contract
                      </Button>
                    </div>
                    <div className="activate-dashboard_button">
                      <Button
                        className="activate-dashboard_button_style"
                        theme="primary"
                      >
                        <Icon icon="plus" />
                        Add Invoice
                      </Button>
                    </div>
                  </Panel>
                </Column>

                <Column width={4}>
                  <Panel title="My tables">
                    {list.map((item, index) => {
                      return (
                        <div className="activate-dashboard_button" key={index}>
                          <Button
                            href={`/tables/${item.name}`}
                            className="activate-dashboard_button_style"
                            theme="primary"
                          >
                            {item.name.replace(/_/g, ' ') }
                          </Button>
                        </div>
                      );
                    })}
                  </Panel>
                </Column>
              </Row>
            </div>
        }
      </div>
    );
  }
}

export default Dashboard;
