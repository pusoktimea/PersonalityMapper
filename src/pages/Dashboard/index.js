import {connect} from 'react-redux';

import {performSearch, resetSearch} from 'actions/search-page';

import {doGetTables} from 'actions/tables-page';

import Dashboard from './Dashboard';

const mapStateToProps = (state) => {
  return {
    isSearchInProgress: state.searchPage.isSearchInProgress,
    isSearchComplete: state.searchPage.isSearchComplete,
    searchResults: state.searchPage.searchResults,
    tables: state.tablesPage.tables,
    isTablesLoaded: state.tablesPage.isTablesLoaded
  };
};

const mapDispatchToProps = (dispatch, {history}) => ({
  onDoSearch: (query) => {
    history.push(`?query=${query}`);
    dispatch(performSearch(query));
  },
  doResetSearch: () => {
    dispatch(resetSearch());
  },
  onGetTables: () => {
    dispatch(doGetTables());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
