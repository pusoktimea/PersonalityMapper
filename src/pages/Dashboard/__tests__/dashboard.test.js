import React from 'react';
import renderer from 'react-test-renderer';
// per https://stackoverflow.com/questions/43771517/using-jest-to-test-a-link-from-react-router-v4
// StaticRouter is needed for the Link component's context
import {StaticRouter} from 'react-router';

import Dashboard from '../Dashboard';

const dashboardProps = {
  isSearchInProgress: false,
  isSearchComplete: false,
  onDoSearch: jest.fn(),
  searchResults: [],
  doResetSearch: jest.fn(),
  location: {},
  isTablesLoaded: true,
  onGetTables: jest.fn(),
  tables: {
    list: [{
      name: 'lorem'
    }, {
      name: 'ipsum'
    }, {
      name: 'dolor'
    }]
  },
  isSideBarMinimised: false
};

describe('Dashboard Page test', () => {
  it('initial load - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <StaticRouter context={{}}>
          <Dashboard {...dashboardProps} />
        </StaticRouter>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('search in progress - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <StaticRouter context={{}}>
          <Dashboard
            {...{
              ...dashboardProps,
              isSearchInProgress: true
            }}
          />
        </StaticRouter>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('search complete - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <StaticRouter context={{}}>
          <Dashboard
            {...{
              ...dashboardProps,
              isSearchComplete: true,
              searchResults: [{
                count: 1,
                table: 'lorem',
                field: 'foo'
              }, {
                count: 20,
                table: 'ipsum',
                field: 'bar'
              }]
            }}
          />
        </StaticRouter>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
