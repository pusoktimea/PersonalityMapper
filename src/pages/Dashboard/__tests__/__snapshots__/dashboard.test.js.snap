// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page test initial load - renders and matches snapshot 1`] = `
<div
  className="activate-dashboard main-content"
>
  <h2
    className="title"
  >
    Dashboard
  </h2>
  <div>
    <div
      className="activate-panel search-page_form"
    >
      <div
        className="activate-panel_body search-page_form"
      >
        <div
          className="activate-input activate-input--light search-page_form_input"
        >
          <input
            className="activate-input_input"
            disabled={false}
            onChange={[Function]}
            placeholder="Search query"
            type="text"
            value=""
          />
        </div>
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled search-page_form_button"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <span
            className="activate-icon fa fa-search"
            id={undefined}
            onClick={undefined}
          />
          Search
        </button>
      </div>
    </div>
    <div
      className="activate-row activate-row--3"
    >
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My reports
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_program"
                onClick={[Function]}
              >
                My Programs
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_goal"
                onClick={[Function]}
              >
                My Goals
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/vendor_report"
                onClick={[Function]}
              >
                Vendor Reports
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My shortcuts
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Programs
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Contract
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Invoice
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My tables
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/lorem"
                onClick={[Function]}
              >
                lorem
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/ipsum"
                onClick={[Function]}
              >
                ipsum
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/dolor"
                onClick={[Function]}
              >
                dolor
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page test search complete - renders and matches snapshot 1`] = `
<div
  className="activate-dashboard main-content"
>
  <h2
    className="title"
  >
    Dashboard
  </h2>
  <div>
    <div
      className="activate-panel search-page_form"
    >
      <div
        className="activate-panel_body search-page_form"
      >
        <div
          className="activate-input activate-input--light search-page_form_input"
        >
          <input
            className="activate-input_input"
            disabled={false}
            onChange={[Function]}
            placeholder="Search query"
            type="text"
            value=""
          />
        </div>
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled search-page_form_button"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <span
            className="activate-icon fa fa-search"
            id={undefined}
            onClick={undefined}
          />
          Search
        </button>
      </div>
    </div>
    <div
      className="activate-panel search-page_results"
    >
      <div
        className="activate-panel_header search-page_results"
      >
        Search Results
      </div>
      <div
        className="activate-panel_body search-page_results"
      >
        <p>
          <strong>
            1
          </strong>
           
          result was
           found in the 
          <strong>
            lorem
          </strong>
           table while looking at the 
          <strong>
            foo
          </strong>
           field.
        </p>
        <p>
          <strong>
            20
          </strong>
           
          results were
           found in the 
          <strong>
            ipsum
          </strong>
           table while looking at the 
          <strong>
            bar
          </strong>
           field.
        </p>
      </div>
    </div>
    <div
      className="activate-row activate-row--3"
    >
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My reports
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_program"
                onClick={[Function]}
              >
                My Programs
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_goal"
                onClick={[Function]}
              >
                My Goals
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/vendor_report"
                onClick={[Function]}
              >
                Vendor Reports
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My shortcuts
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Programs
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Contract
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Invoice
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My tables
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/lorem"
                onClick={[Function]}
              >
                lorem
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/ipsum"
                onClick={[Function]}
              >
                ipsum
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/dolor"
                onClick={[Function]}
              >
                dolor
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page test search in progress - renders and matches snapshot 1`] = `
<div
  className="activate-dashboard main-content"
>
  <h2
    className="title"
  >
    Dashboard
  </h2>
  <div>
    <div
      className="activate-loading"
    >
      <img
        className="activate-loading_full-page-loading"
        src={Object {}}
      />
    </div>
    <div
      className="activate-panel search-page_form"
    >
      <div
        className="activate-panel_body search-page_form"
      >
        <div
          className="activate-input activate-input--light search-page_form_input"
        >
          <input
            className="activate-input_input"
            disabled={false}
            onChange={[Function]}
            placeholder="Search query"
            type="text"
            value=""
          />
        </div>
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled search-page_form_button"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <span
            className="activate-icon fa fa-search"
            id={undefined}
            onClick={undefined}
          />
          Search
        </button>
      </div>
    </div>
    <div
      className="activate-row activate-row--3"
    >
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My reports
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_program"
                onClick={[Function]}
              >
                My Programs
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/reports/my_goal"
                onClick={[Function]}
              >
                My Goals
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/vendor_report"
                onClick={[Function]}
              >
                Vendor Reports
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My shortcuts
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Programs
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Contract
              </button>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <button
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                onClick={undefined}
                to={undefined}
              >
                <span
                  className="activate-icon fa fa-plus"
                  id={undefined}
                  onClick={undefined}
                />
                Add Invoice
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="activate-column activate-column--4"
        width={4}
      >
        <div
          className="activate-panel"
        >
          <div
            className="activate-panel_header"
          >
            My tables
          </div>
          <div
            className="activate-panel_body"
          >
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/lorem"
                onClick={[Function]}
              >
                lorem
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/ipsum"
                onClick={[Function]}
              >
                ipsum
              </a>
            </div>
            <div
              className="activate-dashboard_button"
            >
              <a
                className="activate-button activate-button--primary activate-button--size-default activate-dashboard_button_style"
                disabled={false}
                href="/tables/dolor"
                onClick={[Function]}
              >
                dolor
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
